lib/Semantics/symbol.cpp

```diff
@@ -287,6 +288,14 @@ void Symbol::SetType(const DeclTypeSpec &type) {
       details_);
 }
 
+void Symbol::SetRank(int rank) {
+  std::visit(
+      common::visitors{
+          [&](AssocEntityDetails &x) { x.set_rank(rank); },
+          [](auto &) {},
```

Comment:
Maybe you should crash instead of ignoring the internal error.

You should use `std::get_if<>()` instead of `visit()` when dealing with a single case.

Reply:
How do we crash? Is it the DIE macro?